### DISCUSSION TITLE
Refactoring NetworkScenePostProcssing

### DISCUSSION THREAD

**Discussion by James-Frowen** [MirageNet/Mirage#608] [General]
3 Features of NetworkScenePostProcssing that need solutions:

- [ ] Disable scene objects at build time 
    - [ ] Solution: remove this features
    - [ ] Add to migration guide

- [ ] Find SceneObjects at runtime and spawn them
    - Option 1: Add Component to scene that holds collection of scene objects.
        - started here: https://github.com/MirageNet/Mirage/pull/569
    - Option 2: Find All at runtime, filter by scene

- [ ] Allow SceneObjects to be assigned to a NetworkServer/Client (so they work in mutli-Client setup)

- [ ] Give scene objects a unique scene Id
    - Use OnValidate (or other) to give scene id, fix edge cases)

**Comment by James-Frowen** [comment]
problem: 
> Load the same scene multiple times on the server, to run multiple matches or rooms at once

This is not an issue. 

Server can have multiple objects with the scene Id at once, server only uses scene id to send to client. As long the client doesn't have the same scene open multiple times then the client will know which object it is spawning from the sceneId

**Reply by uweeby** [reply]
If the game requires instancing this could be a slight problem as the server side scene manager currently sends all of the additively loaded scenes to every client for them to load upon first connect.

**Comment by uweeby** [comment]
>Option 2: Find All at runtime, filter by scene

This is sorta what the current default action under NetworkScenePostProcess. It will so a filtered search of all objects in all scenes for any scene objects that need spawning.
